doc/src/guides/solving/solve-equation-algebraically.md

```diff
@@ -0,0 +1,279 @@
+
+```py
+>>> from sympy import pprint, sin, solveset
+>>> from sympy.abc import x
+>>> solution = solveset(sin(x), x)
+>>> pprint(solution)
+{2*n*pi | n in Integers} U {2*n*pi + pi | n in Integers}
+```
+
+However, {func}`~.solve` will return only a finite number of solutions (here, two):
+
+```py
+>>> from sympy import sin, solve
+>>> from sympy.abc import x
+>>> solution = solve(sin(x), x)
+>>> print(solution)
+[0, pi]
+```
+
+### Restricting the domain of solutions using {func}`~.solveset`
+
+By default, SymPy will return solutions in the complex domain, which also includes purely real and imaginary values. Here, the first two solutions are real, and the last two are imaginary:
+
+```py
+>>> from sympy import S, solveset
+>>> from sympy.abc import x
+>>> solution = solveset(x**4 - 256, x)
+>>> print(solution)
+{-4, 4, -4*I, 4*I}
+```
+
+If you want to restrict returned solutions to real numbers, you can specify the [domain to solve in](https://docs.sympy.org/dev/modules/solvers/solveset.html?highlight=solveset#what-is-this-domain-argument-about) as `S.Reals`:
+
+```py
+>>> from sympy import S, solveset
+>>> from sympy.abc import x
+>>> solution = solveset(x**4 - 256, x, domain=S.Reals)
+>>> print(solution)
+{-4, 4}
+```
+
+## Not all equations can be solved
+
+### Equations with no solution
+
+Some equations have no solution, in which case SymPy may return an empty set. For example, the equation $x - 7 = x + 2$ reduces to $-7 = 2$, which has no solution because no value of $x$ will make it true:
+
+```py
+>>> from sympy import solve, Eq
+>>> from sympy.abc import x
+>>> eqn = Eq(x - 7, x + 2)
+>>> solve(eqn)
+[]
+```
+
+So if SymPy returns an empty set, you may want to check whether there is a mistake in the equation.
+
+### Equations with no analytical solution
+
+The vast majority of arbitrary nonlinear equations are not analytically solvable. The classes of equations that are solvable are basically:
+1. Linear equations
+2. Polynomials, except where limited by the Abel-Ruffini theorem
+3. Equations that can be solved by inverting some transcendental functions
+4. Problems that can be transformed into the cases above (e.g., by turning trigonometric functions into polynomials)
+5. A few other special cases that can be solved with something like the Lambert W function
+
+SymPy may reflect that your equation has no solutions that can be expressed algebraically (symbolically) by returning an error such as `NotImplementedError`:
+
+```py
+>>> from sympy import solve, cos
+>>> from sympy.abc import x
+>>> solve(cos(x) - x, x)
+Traceback (most recent call last):
+  ...
+NotImplementedError: multiple generators [x, cos(x)]
+No algorithms are implemented to solve equation -x + cos(x)
+```
+
+so you may have to {func}`solve your equation numerically <sympy.solvers.solvers.nsolve>` instead.
```

Comment:
I think that this alternative suggestion should be presented immediately at the top of this page. For many users this point is the most important point. Right at the beginning we should clarify the distinction between analytic and numeric solutions.